plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env node
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A script that automatically upgrades pieces of the Blockly API
+ *     that have been renamed by modifying developers' local files.
+ */
+'use strict';
+
+import {createAndAddSubCommand} from './command.js';
+import fetch from 'node-fetch';
+import glob from 'glob';
+import {readFile, writeFile} from 'fs';
+import semver from 'semver';
+import JSON5 from 'json5';
+
+const DATABASE_URL = `https://raw.githubusercontent.com/google/blockly/develop/scripts/migration/renamings.json5`;
+
+/**
+ * The version associated with renamings that have not been released yet.
+ * @const {string}
+ */
+const DEV_VERSION = 'develop';
+
+createAndAddSubCommand(
+    'rename', '>=5', 'Perform renamings for breaking changes')
+    .option(
+        '-i [suffix]',
+        'do renamings in-place, optionally create backup files with the ' +
+        'given suffix')
+    .action(async function() {
+      const fromVersion = this.processedArgs[0];
+      const toVersion = this.processedArgs[1];
+      const fileGlobs = this.processedArgs[2];
+      const fileNames = fileGlobs.flatMap((fileGlob) =>
+        glob.sync(fileGlob, {nodir: true, nonull: false}));
```

Comment:
Why are we doing globbing?  Is this for Windows users whose shell doesn't do it for them?  I'd prefer not to glob on platforms where the shell does—it makes it difficult to migrate files that have asterisks in their name…  ;-)